### DISCUSSION TITLE
How to filter on links?

### DISCUSSION THREAD

**Discussion by ghost** [isar/isar#366] [Help] [answered]
In this case, I want to find the product whose variants have at least one stock, it's one-to-many relation.

Here I just map the collection strictly to the remote database model.

The `variant` collection has a field `stock` while the `product` doesn't have.

I am not sure if it's a good practice to add an extra `stock` field to the `product` collection, and calculate the sum when saving to Isar. That's what I did previously.

![Screenshot 2022-05-01 at 13 22 09](https://user-images.githubusercontent.com/8101584/166143803-26ea6d5c-87cc-43b5-991f-3b83d4b2667f.jpg)

**Comment by simc** [comment, marked as answer]
Import the Dart file where you define `Stock`. Then you should see the extension methods for the query 👍
